lib/user_space/user_x86_64_atomic.h

```diff
@@ -38,7 +38,7 @@
  */
 
 struct m0_atomic64 {
-	long a_value;
+	volatile long a_value;
```

Comment:
Why is this needed, please explain.

Reply:
If we do not do this and call m0_atomic64_get() in a loop then gcc with -O2 flag optimizes this statement to read from the register variable instead of reading from the memory location (which it does correctly with -O0 flag).

Reply:
@shashank-parulekar , have you tried adding a memory barrier after the m0_atomic64_get() call to solve this issue? It should be enough to give the compiler a hint, so that it would re-load the variable instead of caching it in the registers.